src/Api-call.js

```diff
@@ -7,12 +7,31 @@ export const retrieveData = async () => {
     const data = await response.json()
     console.log('API Data:', data)
     return(data)
-     
+  
   } catch (error) {
     if (error.message === 'Failed to fetch') {
       throw new Error('Network error: Please check your internet connection')
     }
     console.error(error)
     throw error 
   }
-}
\ No newline at end of file
+}
+
+export const singleMovieId = async (id) => {
+  try {
+    const response = await fetch( `https://rancid-tomatillos.herokuapp.com/api/v2/movies/${id}`)
+    if (!response.ok) {
+      throw new Error("Oops! Something went wrong")
+    }
+    const data = await response.json()
+    console.log('Single Movie Data:', data)
+    return(data)
+
+  } catch (error) {
+    if (error.message === 'Failed to fetch') {
+      throw new Error('Network error: Please check your internet connection')
```

Comment:
How can you save the 'error' in state - the goal being displaying a useful error message to your user in a component?